Release 200.0.0 (#4680)
MetaMask/core: packages/accounts-controller/CHANGELOG.md

## [Unreleased]

## [18.2.0]

### Added

> - Export `AccountsControllerAccountRenamedEvent` type ([#4664](https://github.com/MetaMask/core/pull/4664))
> - Set `nameLastUpdatedAt` metadata when updating account name ([#4589](https://github.com/MetaMask/core/pull/4589))
> - Move logic from `setAccountName` to `updateAccountMetadata` ([#4663](https://github.com/MetaMask/core/pull/4663))
> - Add new event `accountRenamed` ([#4660](https://github.com/MetaMask/core/pull/4660))

Reviewer: WDYT about merging some of the PRs here:

```suggestion
- Add new event `AccountsController:accountRenamed` ([#4664](https://github.com/MetaMask/core/pull/4664)), ([#4660](https://github.com/MetaMask/core/pull/4660))
- Add new `nameLastUpdatedAt` timestamp to account's metadata ([#4589](https://github.com/MetaMask/core/pull/4589)), ([#4663](https://github.com/MetaMask/core/pull/4663))
```

## [Unreleased]

## [18.2.0]

### Added

> - Add new event `AccountsController:accountRenamed` ([#4664](https://github.com/MetaMask/core/pull/4664)), ([#4660](https://github.com/MetaMask/core/pull/4660))

Author: @cryptodev-2s @ccharly 's comment just above suggested to follow another pattern that we did in the past as well. I'm not sure if there's a "definitive" way to do this?

> - Add new event `AccountsController:accountRenamed` ([#4664](https://github.com/MetaMask/core/pull/4664)), ([#4660](https://github.com/MetaMask/core/pull/4660))

Reviewer 2: Sorry I missed that thread, but usually we define the new added event and it's corresponding type you can take as a reference the same CHANGELOG file

> - Add new `nameLastUpdatedAt` timestamp to account's metadata ([#4589](https://github.com/MetaMask/core/pull/4589))

Reviewer: Might be a bit too much? I just feel like we are adding a new field without too much explanations (even though, it's pretty self-explanatory!)

Lemme know what you think!

```suggestion
- Add new `nameLastUpdatedAt` timestamp to account's metadata ([#4589](https://github.com/MetaMask/core/pull/4589))
  - It will be updated every time an account gets renamed through `setAccountName` or `updateAccountMetadata`.
```

### Changed

> - Move logic from `setAccountName` to `updateAccountMetadata` ([#4663](https://github.com/MetaMask/core/pull/4663))

Reviewer 2: ```suugestion

### Changed

- Consolidate `setAccountName` Logic in `updateAccountMetadata` ([#4039](https://github.com/MetaMask/core/pull/4039))
  - Moved the logic for checking account name uniqueness and triggering the `accountRenamed` event from`setAccountName` to `updateAccountMetadata`. The `setAccountName` method now calls`updateAccountMetadata` to handle these tasks.

> - Move logic from `setAccountName` to `updateAccountMetadata` ([#4663](https://github.com/MetaMask/core/pull/4663))

Reviewer 2: what do you think about ?

```suggestion

### Changed

- Consolidate `setAccountName` Logic in `updateAccountMetadata` ([#4039](https://github.com/MetaMask/core/pull/4039))
  - Moved the logic for checking account name uniqueness and triggering the `accountRenamed` event from`setAccountName` to `updateAccountMetadata`. The `setAccountName` method now calls`updateAccountMetadata` to handle these tasks.
 ```

> - Move logic from `setAccountName` to `updateAccountMetadata` ([#4663](https://github.com/MetaMask/core/pull/4663))

Author: Works for me @cryptodev-2s , updated!

## [Unreleased]

## [18.2.0]

### Added

- Add event `AccountsController:accountRenamed` and export corresponding event type `AccountsControllerAccountRenamedEvent` ([#4664](https://github.com/MetaMask/core/pull/4664)), ([#4660](https://github.com/MetaMask/core/pull/4660))
- Add new `nameLastUpdatedAt` timestamp to account's metadata ([#4589](https://github.com/MetaMask/core/pull/4589))

### Changed

> - Consolidate `setAccountName` Logic in `updateAccountMetadata` ([#4663](https://github.com/MetaMask/core/pull/4663))
>   - Moved the logic for checking account name uniqueness and triggering the `accountRenamed` event from`setAccountName` to `updateAccountMetadata`. The `setAccountName` method now calls`updateAccountMetadata` to handle these tasks.

Reviewer 2: ```suggestion
- Consolidate `setAccountName` logic in `updateAccountMetadata` ([#4039](https://github.com/MetaMask/core/pull/4039))
  - Moved the logic for checking account name uniqueness and triggering the `accountRenamed` event from `setAccountName` to `updateAccountMetadata`. The `setAccountName` method now calls `updateAccountMetadata` to handle these tasks.
```
